tests/system_tests/test_api_endpoints.py

```diff
@@ -0,0 +1,190 @@
+        created_tasks.append(created_task)
+
+    task_list = get_response(rest._url("/tasks"), TasksListResponse)
+
+    assert isinstance(task_list, TasksListResponse)
+    task_ids = [task.task_id for task in created_tasks]
+    for task in task_list.tasks:
+        assert task.task_id in task_ids
+        assert task.is_complete is False and task.is_pending is True
+
+    for task in created_tasks:
+        rest.clear_task(task.task_id)
+
+
+def test_get_task_by_id():
+    created_task = rest.create_task(_SIMPLE_TASK)
+
+    get_task = rest.get_task(created_task.task_id)
+    assert (
+        get_task.task_id == created_task.task_id
+        and get_task.is_pending
+        and not get_task.is_complete
+        and len(get_task.errors) == 0
+    )
+
+    rest.clear_task(created_task.task_id)
+
+
+def test_put_worker_task():
+    created_task = rest.create_task(_SIMPLE_TASK)
+    rest.update_worker_task(WorkerTask(task_id=created_task.task_id))
+    active_task = rest.get_active_task()
+    assert active_task.task_id == created_task.task_id
+    rest.clear_task(created_task.task_id)
+
+
+def test_put_worker_task_fails_if_not_idle():
+    small_task = rest.create_task(_SIMPLE_TASK)
+    long_task = rest.create_task(_LONG_TASK)
+
+    rest.update_worker_task(WorkerTask(task_id=long_task.task_id))
+    active_task = rest.get_active_task()
+    assert active_task.task_id == long_task.task_id
+
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        rest.update_worker_task(WorkerTask(task_id=small_task.task_id))
+    assert exception.value.args[0] == "<Response [409]>"
+    time.sleep(1)
+    rest.clear_task(small_task.task_id)
+    rest.clear_task(long_task.task_id)
+
+
+def test_get_worker_state():
+    assert rest.get_state() == WorkerState.IDLE
+
+
+def test_set_state_transition_error():
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        rest.set_state(WorkerState.RUNNING)
+    assert exception.value.args[0] == "<Response [400]>"
+
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        rest.set_state(WorkerState.PAUSED)
+    assert exception.value.args[0] == "<Response [400]>"
+
+
+def test_get_task_by_status():
+    task_1 = rest.create_task(_SIMPLE_TASK)
+    task_2 = rest.create_task(_SIMPLE_TASK)
+    task_by_pending_request = requests.get(
+        rest._url("/tasks"), params={"task_status": TaskStatusEnum.PENDING}
+    )
+    assert task_by_pending_request.status_code == status.HTTP_200_OK
+    task_by_pending = TypeAdapter(TasksListResponse).validate_python(
+        task_by_pending_request.json()
+    )
+
+    assert len(task_by_pending.tasks) == 2
+    for task in task_by_pending.tasks:
+        trackable_task = TypeAdapter(TrackableTask).validate_python(task)
+        assert trackable_task.is_complete is False and trackable_task.is_pending is True
+
+    rest.update_worker_task(WorkerTask(task_id=task_1.task_id))
+    time.sleep(0.1)
```

Comment:
Should: You should really check that the plan is complete rather than just wait 0.1 seconds, that could lead to race conditions and make this test flaky. Again, if you use `BlueapiClient` it should handle this for you.